woocommerce/woocommerce-android
# Prepare carousel for A/B testing

This PR adds the Firebase infrastructure in preparation of A/B testing and loads the Firebase Remote Config parameters on startup. The value is then stored in a DataStore but it's not being used yet as the survey A/B test will be added next sprint. However, I decided to include it because we might need A/B testing functionality for other experiments.

The other change here include:

- The carousel screens have been reordered as per updated designs
- New [carousel action tracking](https://wp.me/pe5sF9-dj#comment-376)
- Matched the onboarding display behavior [with iOS](https://github.com/woocommerce/woocommerce-ios/pull/7307)

**To test:**

- Verify that the screens are now ordered
![image](https://user-images.githubusercontent.com/1522856/180248638-613b3cc6-710f-4e3d-8f3e-038f7dfff593.png)

- Verify that the analytics are being tracked as described by this table:
![image](https://user-images.githubusercontent.com/1522856/180248533-d36be5ad-123f-4020-ad9b-12f924a372b8.png)

- **Display logic test**
_Prerequisite: if you have tested this branch before, please delete and reinstall the app to clear all user defaults._

1. Launch the app
2. Log out if needed --> login onboarding screen should be shown
3. Tap "Next" only twice to go to the last feature page on the onboarding screen so that it is not dismissed
4. Relaunch the app --> login onboarding screen should be shown again
5. Tap "Next" three times or tap "Skip" to go to the prologue screen
6. Relaunch the app --> login onboarding screen should not be shown
7. Log in to the app
8. Log out --> login onboarding screen should not be shown
